cmd/backup-operator/README.md

```diff
@@ -0,0 +1,104 @@
+# etcd backup operator
+
+## Overview
+
+etcd backup operator backups the data of a etcd cluster running on [Kubernetes][Kube] to a remote storage such as AWS [S3][s3]. Currently, backup operator only supports saving backup to AWS S3.
+
+## Requirements
+* Kubernetes 1.7+
+* etcd 3.1+
+
+## Getting Started
+
+You can try out etcd backup operator by running it on `Kubernetes` and then create a `Backup Custom Resource` which contains the targeting etcd cluster and S3 backup config; the etcd backup operator automatically picks up the `Backup Custom Resource`, retrieves etcd data, and then saves data as a backup to S3.
+>Note: The demo uses the `default` kubernete `namespace`.
+
+First, you need to have access to a [Kubernetes][Kube] environment either locally via [Minikube][minikube] or any Cloud Providers that provide `Kubernetes` as a service.
+
+Once you have access to a `Kubernetes` setup, you need to create a [etcd operator][etcd_operator] pod in order to deploy an etcd cluster. See [instructions on how to install/uninstall etcd operator][etcd_operator_install] for more detail.
+
+Next, use etcd operator to create a example etcd cluster:
+
+```sh
+$ kubectl create -f example/example-etcd-cluster.yaml
+$ kubectl get po
+NAME                            READY     STATUS    RESTARTS   AGE
+etcd-operator-857677187-9s6x7   1/1       Running   0          1m
+example-etcd-cluster-0000       1/1       Running   0          28s
+example-etcd-cluster-0001       1/1       Running   0          19s
+example-etcd-cluster-0002       1/1       Running   0          10s
+```
+
+Once `example-etcd-cluster` is running, you can create a backup using etcd backup operator. First, deploy an etcd backup operator:
+> Note: etcd backup operator also creates Backup CRD is created when starting.
+
+```sh
+$ kubectl create -f example/etcd-backup-operator/deployment.yaml
+$ kubectl get po
+NAME                                    READY     STATUS    RESTARTS   AGE
+etcd-backup-operator-1102130733-hhgt7   1/1       Running   0          3s
+...
+```
+
+Then, you need to create an kubernetes secret that contains aws config/credential; etcd operator uses the secret to gain access to S3 in order to save the etcd backup.
+
+You need to have aws config and credentials files saved locally:
+```sh
+$ cat $AWS_DIR/credentials
+[default]
+aws_access_key_id = XXX
+aws_secret_access_key = XXX
+
+$ cat $AWS_DIR/config
+[default]
+region = <region>
+```
+
+Create kubernetes secret `aws`:
+
+`$ kubectl create secret generic aws --from-file=$AWS_DIR/credentials --from-file=$AWS_DIR/config`
+
+Finally, create a Backup CR `backup_cr.yaml` to trigger a backup:
+```yaml
+apiVersion: "etcd.database.coreos.com/v1beta2"
```

Comment:
Please use `example/etcd-backup-operator/backup_cr.yaml` .

mention that what needs to be modified.